acceptance/internal/bundle_config_test.go

```diff
@@ -0,0 +1,162 @@
+	"gopkg.in/yaml.v3"
+)
+
+func TestMergeBundleConfig(t *testing.T) {
+	tests := []struct {
+		name         string
+		initialYaml  string
+		bundleConfig map[string]any
+		expected     map[string]any
+	}{
+		{
+			name:         "new_file",
+			initialYaml:  "",
+			bundleConfig: map[string]any{},
+			expected:     map[string]any{},
+		},
+		{
+			name:         "empty_config",
+			initialYaml:  "{}",
+			bundleConfig: map[string]any{},
+			expected:     map[string]any{},
+		},
+		{
+			name:        "simple_top_level",
+			initialYaml: "{}",
+			bundleConfig: map[string]any{
+				"name": "test-bundle",
+			},
+			expected: map[string]any{
+				"name": "test-bundle",
+			},
+		},
+		{
+			name:        "simple_top_level_new_file",
+			initialYaml: "",
+			bundleConfig: map[string]any{
+				"name": "test-bundle",
+			},
+			expected: map[string]any{
+				"name": "test-bundle",
+			},
+		},
+		{
+			name:        "nested_config",
+			initialYaml: "{}",
+			bundleConfig: map[string]any{
+				"bundle": map[string]any{
+					"name": "test-bundle",
+				},
+			},
+			expected: map[string]any{
+				"bundle": map[string]any{
+					"name": "test-bundle",
+				},
+			},
+		},
+		{
+			name:        "nested_config_new_file",
+			initialYaml: "",
+			bundleConfig: map[string]any{
+				"bundle": map[string]any{
+					"name": "test-bundle",
+				},
+			},
+			expected: map[string]any{
+				"bundle": map[string]any{
+					"name": "test-bundle",
+				},
+			},
+		},
+		{
+			name: "merge_with_existing_config",
+			initialYaml: `
+bundle:
+  name: original-name
+  target: dev
+`,
+			bundleConfig: map[string]any{
+				"bundle": map[string]any{
+					"name": "default-name",
+				},
+			},
+			expected: map[string]any{
+				"bundle": map[string]any{
+					"name":   "original-name",
+					"target": "dev",
+				},
+			},
+		},
+		{
+			name:        "merge_with_existing_config_2",
+			initialYaml: `resources: {}`,
+			bundleConfig: map[string]any{
+				"bundle": map[string]any{
+					"name": "new-name",
+				},
+			},
+			expected: map[string]any{
+				"bundle": map[string]any{
+					"name": "new-name",
+				},
+				"resources": map[string]any{},
+			},
+		},
+
+		{
+			name: "multiple_nested_levels",
+			initialYaml: `
+resources:
+  jobs:
+    myjob:
+      hello: world
+  pipelines:
+    mypipeline:
+      name: 123
+`,
+			bundleConfig: map[string]any{
+				"resources": map[string]any{
+					"jobs": map[string]any{
+						"myjob": map[string]any{
+							"name": "My Job",
+						},
+					},
+				},
+			},
+			expected: map[string]any{
+				"resources": map[string]any{
+					"jobs": map[string]any{
+						"myjob": map[string]any{
+							"name":  "My Job",
+							"hello": "world",
+						},
+					},
+					"pipelines": map[string]any{
+						"mypipeline": map[string]any{
+							"name": 123,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			out, err := MergeBundleConfig(tt.initialYaml, tt.bundleConfig)
```

Comment:
Can this function take a `map[string]any` as input instead?

Reply:
it already does?

> bundle_config.go:func MergeBundleConfig(source string, bundleConfig map[string]any) (string, error) {

Reply:
I mean for "source". The mix of unmarshalling and receiving an object is confusing.

Reply:
I see what you mean, but there is some logic to it: the function receives config as a string and returns new updated config as a string. This allows it to encapsulate (and potentially have test cases for) details related to yaml parsing, such as whether it is strict or not, whether it preserves comments. The test runner on the other hand knows which file we're reading and writing and handles all I/O.

The fact that it also receives unmarshalled bundleConfig is just because we do all of unmarshalling of config in one place and there is no other way to receive bundleConfig.

Reply:
There is a lot of unnecessary marshalling / unmarshalling going on with this implementation though. It works for now and does not cause noticeable overhead, but worth rewriting a bit (for later).